apps/api/src/services/events.service.ts

```diff
@@ -0,0 +1,147 @@
+import { Injectable } from '@nestjs/common';
+import {
+	CreateEventDto,
+	EventParams,
+	UpdateEventDto,
+} from '@api/dto/events.dto';
+import { EventEmitter2 } from '@nestjs/event-emitter';
+import { CommentModel, EventModel, FormModel, KanbanModal } from '@api/models';
+import { User } from '@api/db/schema';
+import { stripUndefinedOrNull, exclude } from '@api/utils';
+import { FFError } from '@api/utils/error';
+
+@Injectable()
+export class EventsService {
+	constructor(private readonly eventEmitter: EventEmitter2) {}
+
+	public async getEventById(id: string) {
+		return await EventModel.findById(id);
+	}
+
+	async createEvent({
+		newEvent,
+		user,
+		orgId,
+	}: {
+		newEvent: CreateEventDto;
+		user: User;
+		orgId: string;
+	}) {
+		const isEventWithSlugExist = await EventModel.find({
+			slug: newEvent.name,
+		});
+
+		let slug: string;
+
+		if (isEventWithSlugExist) {
+			const totalEventCount = await EventModel.count({
+				slug: newEvent.name,
+			});
+			slug = `${newEvent.name}-${totalEventCount}`;
+		} else {
+			slug = newEvent.name;
+		}
+
+		const event = await EventModel.insert({
+			fk_organization_id: orgId,
+			name: newEvent.name,
+			website: newEvent.website,
+			location: newEvent.location,
+			slug,
+			event_date: newEvent.event_date,
+			description: newEvent.description ?? '',
+		});
+
+		// Creating an Event will create a default Form
+		await FormModel.insert({
+			fk_event_id: event.id,
+			title: 'Default Form',
+			description: 'Default Form Description',
+			confirmation_message: 'This message is shown after form submission',
+			misc: {},
+			is_default_form: true,
+		});
+
+		// Creating an Event will create a Kanban board
+		await KanbanModal.insertMany([
+			{
+				fk_event_id: event.id,
+				title: 'ToDo',
+			},
+			{
+				fk_event_id: event.id,
+				title: 'In Progress',
+			},
+			{
+				fk_event_id: event.id,
+				title: 'Done',
+			},
+		]);
+
+		this.eventEmitter.emit('event.created', event, user.id);
+	}
```

Comment:
can we convert this to event emitter ?

Reply:
let's keep as this now. Will be more predictable this way.